use-cases/virgo/trainer.py

```diff
@@ -26,30 +25,33 @@ def __init__(
         batch_size: int,
         learning_rate: float = 1e-3,
         num_epochs: int = 2,
+        config: Union[Dict, TrainingConfiguration] | None = None,
         generator: Literal["simple", "deep", "resnet", "unet"] = "unet",
         loss: Literal["L1", "L2"] = "L1",
-        strategy: Literal["ddp", "deepspeed", "horovod"] = 'ddp',
+        strategy: Literal["ddp", "deepspeed", "horovod"] | None = 'ddp',
```

Comment:
This should use Optional instead of "| None" and since it has a default value it would need to be explicitly set to None, i.e. "strategy=None", for this to actually be None, so consider whether this even is Optional or not.